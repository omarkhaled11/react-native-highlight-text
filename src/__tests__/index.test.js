import React from 'react';
import { Text } from 'react-native';
import { create } from 'react-test-renderer';

import HighlightText from '../';

describe('<HighlightText />', () => {
  it('should render properly', () => {
    const wrapper = create(
      <HighlightText
        searchWords={['hello', 'world']}
        textToHighlight={'In this text we can do a hello to the entire world'}
      />
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('should render properly with custom styles', () => {
    const wrapper = create(
      <HighlightText
        searchWords={['hello', 'world']}
        textToHighlight={'In this text we can do a hello to the entire world'}
        style={{ color: 'darkgray' }}
        highlightStyle={{ color: 'blue' }}
      />
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('should render properly with custom components', () => {
    const CustomText = (props) => (
      <Text style={{ color: '#333' }}>{props.children}</Text>
    );
    const CustomHighlight = (props) => (
      <Text style={{ fontWeight: 'bold' }}>{props.children}</Text>
    );
    const wrapper = create(
      <HighlightText
        searchWords={['hello', 'world']}
        textToHighlight={'In this text we can do a hello to the entire world'}
        textComponent={CustomText}
        highlightComponent={CustomHighlight}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
