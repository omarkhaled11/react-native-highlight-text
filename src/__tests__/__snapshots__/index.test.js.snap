// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HighlightText /> should render properly 1`] = `
<Text>
  In this text we can do a 
  <Text>
    hello
  </Text>
   to the entire 
  <Text>
    world
  </Text>
</Text>
`;

exports[`<HighlightText /> should render properly with custom components 1`] = `
<Text
  style={
    Object {
      "color": "#333",
    }
  }
>
  In this text we can do a 
  <Text
    style={
      Object {
        "fontWeight": "bold",
      }
    }
  >
    hello
  </Text>
   to the entire 
  <Text
    style={
      Object {
        "fontWeight": "bold",
      }
    }
  >
    world
  </Text>
</Text>
`;

exports[`<HighlightText /> should render properly with custom styles 1`] = `
<Text
  style={
    Object {
      "color": "darkgray",
    }
  }
>
  In this text we can do a 
  <Text
    style={
      Object {
        "color": "blue",
      }
    }
  >
    hello
  </Text>
   to the entire 
  <Text
    style={
      Object {
        "color": "blue",
      }
    }
  >
    world
  </Text>
</Text>
`;
